File: python/project/src/functions/passthrough.py
#!/usr/bin/env python3
"""
Pulsar Function that returns the input message. Used for decoupling flows and creating stop valves.
This function can be deployed by simply uploading just this file. 
(No need to zip it for deployment since it has no dependencies outside of pulsar.)
"""
import pulsar

class Passthrough(pulsar.Function):
    def __init__(self):
        self.userConfig = None
        self.logger = None
        self.isInitialized = False
    def initialize(self, context):
        self.logger = context.get_logger()
        self.logger.info("Context is: {}".format(context))
        self.userConfig = context.get_user_config_map()
        self.isInitialized = True
    def process(self, message, context):
        if self.isInitialized == False:
            self.initialize(context)
        return message

File: java/incident-passthrough/src/main/java/com/trafficcorp/example/demofunctions/function/Passthrough.java
package com.trafficcorp.example.demofunctions.function;

import org.apache.pulsar.functions.api.Context;
import org.apache.pulsar.functions.api.Function;

public class Passthrough implements Function<byte[], byte[]> {
    public Passthrough() {

    }
    @Override
    public byte[] process(byte[] input, Context context) throws Exception {
        return input;
    }
}